src/reducers/dataReducer.jsx

```diff
@@ -8,8 +8,55 @@ export const dataReducer = (state, action) => {
     case "FETCH_PRODUCTS":
       return {
         ...state,
-        products: [...action.payload]
+        products: [...action.payload],
+        isProducts:true
       };
+    case "PRICE_HIGH_TO_LOW":
+      return{
+        ...state,
+        sortByPrice:action.payload
+      }
+      case "PRICE_LOW_TO_HIGH":
+        return{
+          ...state,
+          sortByPrice:action.payload
+        }
+      case "SET_FILTERED_PRODUCTS":
+        return{
+          ...state,
+          filteredProducts:[...action.payload],
+          isFilteredProducts:true
+        }
+      case "FILTER_BY_RATING":
+        return{
+          ...state,
+          filterByRatings:Number(action.payload)
+        }
+      case "CATEGORY_ACTION_FIGURES":
+        return{
+          ...state,
+          categoryActionFigures:action.payload
+        }
+        case "CATEGORY_T_SHIRTS":
+          return{
+            ...state,
+            categoryTshirts:action.payload
+          }
+        case "CATEGORY_MASKS":
+          return{
+            ...state,
+            categoryMasks:action.payload
+          }
+        case "FILTERS_RESET":
```

Comment:
for filters reset keep one initialState Object as cartState={} as constant in this filter reset do return {...state,cartState} it will be much better

Reply:
Okay got it.